src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -62,9 +62,20 @@ public WebScriptHostManager(ScriptHostConfiguration config, ISecretManagerFactor
                 config.TraceWriter = systemTraceWriter;
             }
 
-            _secretManager = secretManagerFactory.Create(settingsManager, config.TraceWriter, new FileSystemSecretsRepository(webHostSettings.SecretsPath));
             _performanceManager = new HostPerformanceManager(settingsManager);
             _swaggerDocumentManager = new SwaggerDocumentManager(config);
+            
+            var secretStorageType = settingsManager.GetSetting("UseBlobSecretStorage"); //TODO: official parameter naming                        
+            if (secretStorageType != null) 
```

Comment:
For now, it would also be safer to check both conditions, 1) that blob storage for secrets is enabled and 2) we have a storage connection string.